app/components/provider_interface/course_summary_component.rb

```diff
@@ -0,0 +1,35 @@
+module ProviderInterface
+  class CourseSummaryComponent < ViewComponent::Base
+    attr_reader :application_choice, :provider_name, :course_name_and_code,
+                :location_name_and_address, :study_mode
+
+    def initialize(application_choice:)
```

Comment:
Can we make this more specific and use course instead? `application_choice.provider` is retrieved through the course model anyway, similar to course_option.

Reply:
I was attempting to pass in the course instead, but realised because we are surfacing the study mode we do need the course option as using the study mode on the course itself won't give us the definitive result we need (it come back as both full time or part time if those exist on the course options). This means we will need to pass the course option itself here, but it feels a bit disjoined without the application choice and might be better to go with the application choice, let me know your thoughts around keeping it as is or using the `course_option` instead.

Reply:
I would still go for the object closer to matching all dependencies, `course_option` as you correctly mentioned rather than `course`. I always get confused about which level they live on, course and course_option! 
Which makes the component reusable for the offered course (or rather offered_course_option) as well.